data-in-pipeline/app/flow.py

```diff
@@ -4,22 +4,56 @@
 from app.identify.main import identify_source_document
 from app.load.rds import load_rds
 from app.transform.main import transform
+from app.upload.aws_bucket import upload_to_s3
 
 
-@flow()
-def process_document_updates(ids: list[str] = []):
-    result = document_pipeline.map(ids)
-    return [result.result().id for result in result]
+@task(log_prints=True)
+def extract_document(document_id: str):
+    """Extract document from Navigator."""
+    return extract_navigator_document(document_id)
 
 
 @task(log_prints=True)
-def document_pipeline(id: str):
-    navigator_document = extract_navigator_document(id)
+def cache_document(navigator_document):
+    """Upload raw document to S3 cache."""
+    upload_to_s3(
+        navigator_document.model_dump_json(),
+        bucket="cpr-production-document-cache",
+        key=f"navigator/{navigator_document.data.import_id}.json",
+    )
 
-    identified_source_document = identify_source_document(navigator_document)
 
-    document = transform(identified_source_document)
+@task(log_prints=True)
+def identify_document(navigator_document):
+    """Identify source document type."""
+    return identify_source_document(navigator_document)
+
+
+@task(log_prints=True)
+def transform_document(identified_source_document):
+    """Transform document to target format."""
+    return transform(identified_source_document)
+
 
+@task(log_prints=True)
+def load_document(document):
+    """Load document to RDS."""
     load_rds(document)
+    return document
+
 
+@task(log_prints=True)
+def document_pipeline(id: str):
+    """Process a single document through the pipeline."""
+    navigator_document = extract_document(id)
+    cache_document(navigator_document)
```

Comment:
my understanding is that this PR is meant to be implementing the `load` step - is this right?

or are we creating caches between the steps?

I think we should go for the former, and then think of a caching strategy, but what you have here feels like it would work nicely.

Reply:
@odrakes-cpr question - why are we caching at this point?

My understanding of this PR is to have the load step in ETL?

i.e. after the transformer load the transformed document into S3.

I am still unclear as to what this cache step is?